apps/ec.c

```diff
@@ -219,6 +219,9 @@ int ec_main(int argc, char **argv)
 
     if (no_public)
         EC_KEY_set_enc_flags(eckey, EC_PKEY_NO_PUBKEY);
```

Comment:
Shouldn't the if-part also just modify a single bit? If not for symmetry, then maybe for robustness?  Currently, the only other flag is `EC_PKEY_NO_PARAMETERS`, which apparently is not set when loading the key. But this may change in the future. 

```C
    if (no_public)
        EC_KEY_set_enc_flags(eckey,
                             EC_KEY_get_enc_flags(eckey) | EC_PKEY_NO_PUBKEY);
    else
        EC_KEY_set_enc_flags(eckey,
                             EC_KEY_get_enc_flags(eckey) & ~EC_PKEY_NO_PUBKEY);
```
@mattcaswell, what is your opinion?

Reply:
Makes sense to me.

Reply:
OK understood but is this related to this pull-request ? If you think it is, I can edit the commit again.
An idea would be to have EC_KEY_set_enc_flag / EC_KEY_unset_enc_flag instead of EC_KEY_set_enc_flags, which would be easier to use.

Reply:
I like your idea to add a reusable function for that. If you prefer it, you can create two separate commits for this pr, one that adds the new functions and the other that fixes your issue.

Reply:
Please remember to update the documentation accordingly in the commit that adds the new functions.

Reply:
Before you implement it, please have a look at existing implementations (`grep -r -e _set_flags -e _clear_flags crypto`): They all have the following form:

```
void EC_KEY_set_flags(EC_KEY *key, int flags)
{
    key->flags |= flags;
}

void EC_KEY_clear_flags(EC_KEY *key, int flags)
{
    key->flags &= ~flags;
}
```

Unfortunately, `EC_KEY_set_enc_flags()` violates this principle, and this can't be changed :-/. But at least you could replace "unset" by "clear" for consistency: `EC_KEY_set_enc_flag / EC_KEY_clear_enc_flag`

Reply:
I pushed new commits on my branch, and amend the existing one.
Can you please confirm me this is the right way to update PR ?